CHANGELOG.md

```diff
@@ -42,6 +42,7 @@ This section is for maintaining a changelog for all breaking changes for the cli
 ### Dependencies
 
 ### Changed
+- Changed SearchAfter of SearchRequest type to FieldValue instead of String
```

Comment:
Add a link to the PR like other line-items.

Reply:
Since overload cannot distinguish between List<FieldValue> and List<String>, I think you have no choice but to make backwards incompatible. It can be implemented as shown below, but it seems that there will be a side effect of allowing unintended Objects.

And by adding this overloading, the issue of not being able to find the targeted method in the deserializer also occurs.

<img width="536" alt="image" src="https://github.com/opensearch-project/opensearch-java/assets/102130859/35fa1b82-a303-473c-9bda-d0b7477d0fbf">

<img width="885" alt="image" src="https://github.com/opensearch-project/opensearch-java/assets/102130859/8836e681-6655-4f55-88ce-972d33f964ef">

Reply:
@reta any ideas what's best here?

Reply:
@reta Can you give me any comment here? 🥲

Reply:
Sorry folks, overlooked this pull request, we could add the varargs overload:
```
public final Builder searchAfter(String value, String... values) {
```

but indeed due to erasure, we may only keep  `List<String>` for backward compatibility in 2.x (and convert it to `FielValue` internally) but use  `List<FieldValue>` in 3.x, what do you think?

Reply:
@reta Converting to FieldValue only internally doesn't seem to make sense. So, how about applying this PR to 3.x? If I have to do that, should I move the branch to 3.x?

Reply:
Sure @channel-dante , the `main` is 3.x so no need to change anything